shared/js/background/dynamic-rule-id.js

```diff
@@ -0,0 +1,40 @@
+const browserWrapper = require('./wrapper.es6')
+const DYNAMIC_RULE_ID_START = 100000
+const DYNAMIC_RULE_STORAGE_KEY = 'dynamicRuleOffset'
+let dynamicRuleOffset
+
+export function getDynamicRuleId () {
+    const nextRuleId = DYNAMIC_RULE_ID_START + dynamicRuleOffset
+    dynamicRuleOffset += 1
+    browserWrapper.setToSessionStorage(DYNAMIC_RULE_STORAGE_KEY, dynamicRuleOffset)
+    return nextRuleId
+}
+
+async function _getExistingRulesAssigned () {
+    let rules = []
+
+    // get all declarative rules
+    const set1 = await chrome.declarativeNetRequest.getDynamicRules()
+    rules = rules.concat(set1.filter(r => r.id >= DYNAMIC_RULE_ID_START))
+    const set2 = await chrome.declarativeNetRequest.getSessionRules()
+
+    // get all rules that were already dynamically created
+    rules = rules.concat(set2.filter(r => r.id >= DYNAMIC_RULE_ID_START))
+    return rules
+}
+
+// set offset and update based on rules that have been created
+(async () => {
+    // get existing offset in storage
+    dynamicRuleOffset = await browserWrapper.getFromSessionStorage(DYNAMIC_RULE_STORAGE_KEY) || 0
```

Comment:
I figure that if reading the existing offset from session storage is successful, we should just use that. If reading wasn't successful _then_ we want to figure out what the offset is the hard way. I figure that's kind of the whole point of the session variable, to avoid us having to constantly re-read all the DNR rules every time the background ServiceWorker restarts.